src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java

```diff
@@ -133,14 +133,13 @@ public HttpResponse doCheckFileId(StaplerRequest req, @AncestorInPath Item conte
             // allowed to you
             if (context != null) {
                 context.checkPermission(Item.CONFIGURE);
+                final Config config = ConfigFiles.getByIdOrNull(context, fileId);
+                if (config != null) {
+                    return ConfigFileDetailLinkDescription.getDescription(req, context, fileId);
+                }
             }
-            
-            final Config config = ConfigFiles.getByIdOrNull(context, fileId);
-            if (config != null) {
-                return ConfigFileDetailLinkDescription.getDescription(req, context, fileId);
-            } else {
-                return FormValidation.error("you must select a valid file");
-            }
+
+            return FormValidation.error("you must select a valid file");
```

Comment:
but doesn't `context` == `null` in the pipeline snippet generator - thus this will always fail in that context?

Reply:
Yes, it will always display the message that you must select a file, but it does not fail with a null pointer exception

Reply:
In the end, the `doFillFileIdItems` is already bypassing the permission check when it's called outside of a project.